### ISSUE TITLE
Hyperlink is not working

### ISSUE THREAD

**Issue by DeekshaPrabhub96** [PHPOffice/PhpSpreadsheet#2389]
This is: I have an excel where link is added to a cell. The link points to another sheet of same file and  I am opening an excel file and updating a cell value with text and saving it as different file.

$spreadsheet = \PhpOffice\PhpSpreadsheet\IOFactory::load($filePath);
$spreadsheet->getCalculationEngine()->suppressFormulaErrors=true;
.
.
.
$writer = new Xlsx($spreadsheet);
$writer->save('output.xlsm');

### What is the expected behavior?
Hyper link should be preset in the new file.

### What is the current behavior?
Hyper links are not carried forward.

### Which versions of PhpSpreadsheet and PHP are affected? 

I have attached the sample input and output files, please refer sheet2 for hyper link.
 
[HyperLinkOutput.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/7511656/HyperLinkOutput.xlsx)
[HyperLinkInput.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/7511659/HyperLinkInput.xlsx)

**Comment by DeekshaPrabhub96** [comment]
Hello,

Thanks this fixed the issue for hyperlink.
I observed that whenever there are hyperlinks and dropdown lists (Datavalidations List) in same sheet, it fails and when I try to open the new file it enters recovery mode.

I have attached the sample input and output files. Sheet2 A1 has a hyperlink and B1 has a datavalidation List.

[HyperLinkInput (1).xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/7520555/HyperLinkInput.1.xlsx)
[HyperLinkOutput (1).xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/7520556/HyperLinkOutput.1.xlsx)
